File: 2439_minimizeMaximumOfArray/main.cpp
#include <vector>
#include <algorithm>
#include <cmath>

class Solution {
public:
    int minimizeArrayValue(std::vector<int>& nums) {
        if(nums.size() == 0)
            return 0;
        long res = nums[0];
        long total = nums[0];
        for(int i = 1; i < nums.size(); ++i) {
            total = total + nums[i];
            res = std::max(res, (long)std::ceil((double)total / (i+1)));
        }
        return res;
    }
};

File: 2439_minimizeMaximumOfArray/main.py
import math

class Solution:
    def minimizeArrayValue(self, nums: List[int]) -> int:
        if len(nums) == 0:
            return 0
        
        res = nums[0]
        total = nums[0]
        for i in range(1, len(nums)):
            total += nums[i]
            res = max(res, math.ceil(total / (i+1)))
        return res